lib/ecstatic.js

```diff
@@ -330,6 +339,19 @@ var ecstatic = module.exports = function (dir, options) {
 ecstatic.version = version;
 ecstatic.showDir = showDir;
 
+function hasGzipId12(gzipped, cb) {
+  var stream = fs.createReadStream(gzipped, { start: 0, end: 1, autoClose: true }),
+      isGzip = false;
+
+  stream.on("close", function () {
+    cb(isGzip);
+  });
+
+  stream.on("data", function (chunk) {
```

Comment:
looking at concat-stream it looks like its primary differences are around encoding--if we only work with buffers and nothing else, then this is mostly ok.

The one thing you're missing is a stream error handler, and protection against double-callback-ing and other corner cases. I also run travis tests against 0.12, which affects what buffer methods you can use. So something more like,

```js
var stream = fs.createReadStream(gzipped, { start: 0, end: 1, autoClose: true }),
    buffer = Buffer(''),
    hasBeenCalled = false;

stream.on('data', function (chunk) {
  buffer = Buffer.concat([buffer, chunk], 2)
});

stream.on('error', function (err) {
  if (hasBeenCalled) {
    throw err;
  }
  hasBeenCalled = true;
  cb(err);
});

stream.on('close', function () {
  if (hasBeenCalled) {
    return;
  }
  hasBeenCalled = true;
  if (buffer.length !== 2) {
    return cb(null, false);
  }
  return cb(null, buffer[0] == 31 && buffer[1] == 139);
});
```